### PR TITLE
Add country field for customer

### LINKED ISSUE #246: Adding country to customer properties
Currently it is not possible to enter a specific country for a customer.
This is required, when you have customers in different countries

### PR DESCRIPTION
**Pull request by simonschaufi**
FIXES #246

### CODE REVIEW

File: installer/install.php [hunk 1]
```diff
@@ -175,6 +175,7 @@ function quoteForSql($input) {
   `street` varchar(255) NULL,
   `zipcode` varchar(255) NULL,
   `city` varchar(255) NULL,
+  `country` varchar(255) NULL,
```
Comment (kevinpapst) [inline installer/install.php]: This is different from "DEFAULT NULL" in the updater below, but it might be that MySql handles it the same way.

Comment (simonschaufi) [inline installer/install.php]: This is correct mysql syntax. You could even leave the NULL out.

Comment (kevinpapst) [inline installer/install.php]: I know that. My point is: do not rely on implicit values. Define "default null" either twice or not at all,.

Comment (simonschaufi) [inline installer/install.php]: for now let's just be consistent and do it "properly" in a separate commit.

File: updater/updater.php [hunk 2]
```diff
@@ -1117,6 +1117,11 @@
 
 // release of kimai 1.0.0
 
+if ((int)$revisionDB < 1388) {
```
Comment (kevinpapst) [inline updater/updater.php]: don't forget to change this after merging

**Comment by kevinpapst** [comment]
A general thought here: A free text field is useful for rendering in addresses/invoices. 
But for future features, a select field with a ISO code (should be available through the Zend Framework) might be better solution. Then we could probably fetch formats and currency automatically.

File: libraries/Kimai/Invoice/AbstractRenderer.php [hunk 3]
```diff
@@ -124,12 +124,15 @@ public function canRender()
     
     protected function prepareCustomerArray($customer)
     {
+        global $kga;
```
Comment (simonschaufi) [inline libraries/Kimai/Invoice/AbstractRenderer.php:127]: I wasn't sure here if this is the way to go or if there is a better solution?

Comment (kevinpapst) [inline libraries/Kimai/Invoice/AbstractRenderer.php:127]: No, not until #688 is merged

Comment (simonschaufi) [inline libraries/Kimai/Invoice/AbstractRenderer.php]: ok then let me check the invoice again and then this is ready to merge!

**Comment by kevinpapst** [comment]
Cool. Haven't tried it, but looks cool!
Now the invoice renderer knows the two-letter country code, right? Does it know a translated country name as well? This must not be necessary part of this PR, but it would be cool.

**Comment by simonschaufi** [comment]
Of course i render the full country name. Nobody wants iso codes in an invoice. check the lines below the "global"

ok ready for merge!

File: libraries/Kimai/Invoice/AbstractRenderer.php [hunk 4]
```diff
@@ -124,12 +124,15 @@ public function canRender()
...
+        $countryName = Zend_Locale::getTranslation($customer['country'], 'country', $kga['language']);
```
Comment (kevinpapst) [inline libraries/Kimai/Invoice/AbstractRenderer.php]: What happens if $customer['country'] is empty?
Should that line be included in line 135 in the isset() check?

Comment (simonschaufi) [inline libraries/Kimai/Invoice/AbstractRenderer.php]: no need. it will just return nothing.

**Pull request merged**